### ISSUE TITLE
plan to support fabric new architecture 

### ISSUE THREAD

**Issue by chanphiromsok** [livekit/client-sdk-react-native#134]
React Native new Arch here is the libs status https://docs.google.com/spreadsheets/d/1F1tI9PQLl_uab3HNYNwgjPVL2r0hJ9VymXcgDp5etdg/edit?pli=1#gid=1521034944

**Comment by davidliu** [comment]
https://github.com/reactwg/react-native-new-architecture/blob/main/docs/enable-libraries-prerequisites.md

- [x] Interop layer compatibility
- [ ] [Define Specs in JavaScript](https://github.com/reactwg/react-native-new-architecture/blob/main/docs/enable-libraries-prerequisites.md#define-specs-in-javascript)
- [ ] [Configure Codegen](https://github.com/reactwg/react-native-new-architecture/blob/main/docs/enable-libraries-prerequisites.md#configure-codegen)
- [ ] [Migrate off UIManager JavaScript APIs](https://github.com/reactwg/react-native-new-architecture/blob/main/docs/enable-libraries-prerequisites.md#migrate-off-uimanager-javascript-apis)

- [ ] [Android Compatibility](https://github.com/reactwg/react-native-new-architecture/blob/main/docs/enable-libraries-android.md)
- [ ] [iOS Compatibility](https://github.com/reactwg/react-native-new-architecture/blob/main/docs/enable-libraries-ios.md)

**Comment by chanphiromsok** [comment]
do you have plan to support new arch ?

**Comment by davidliu** [comment]
Yeah.

**Comment by chanphiromsok** [comment]
> Yeah.

that great

**Comment by radko93** [comment]
New architecture is enabled by default in the upcoming RN release.

**Comment by Jonatthu** [comment]
Any updates on this?

**Comment by davidliu** [comment]
@Jonatthu We heavily depend on react-native-webrtc, which hasn't completed its migration to the new architecture. This probably won't move to the new architecture until that migration is completed first.

You can track the progress here: https://github.com/react-native-webrtc/react-native-webrtc/pull/1590

The interop layer should still work with the new architecture in the meantime.

**Comment by radko93** [comment]
Everything works just fine with the Interop layer.

**Comment by davidliu** [comment]
> Do anyone think, we should be working towards pure new arch rather than interop layer... Will we see performance differences if we move to pure new arch?
> 
> [@radko93](https://github.com/radko93) [@davidliu](https://github.com/davidliu)

@Arjit0762 It's unlikely that there will be any significant difference if we moved to pure new architecture. The overhead of the interop layer is not significant enough for the amount of calls made.

**Comment by davidliu** [comment]
@shaima-shafeer yes, the library works when using the new architecture. There's no extra steps needed. If you're unable to connect, file a separate issue.

**Comment by shaima-shafeer** [comment]
> [@shaima-shafeer](https://github.com/shaima-shafeer) yes, the library works when using the new architecture. There's no extra steps needed. If you're unable to connect, file a separate issue.

https://github.com/livekit/client-sdk-react-native/issues/305
